lib/cachex/services/courier.ex

```diff
@@ -76,7 +76,13 @@ defmodule Cachex.Services.Courier do
               rescue
                 e -> { :error, Exception.message(e) }
               end
-            send(parent, { :notify, key, result })
+            # Commit the result immediately to avoid a race in fetch
```

Comment:
You can drop this comment; I'll address it later in the doc comments.

Reply:
OK, will do